.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+          S3_SYNC_OPTS: "--dry-run --recursive --API-ACL=${{ steps.s3-acl-private.outputs.string || steps.s3-acl-public.outputs.string }}"
+          S3_URL: "s3://${{ env.S3_BUCKET }}/${{ steps.s3-images-dir.outputs.string || steps.s3-esr-images-dir.outputs.string }}"
+          SLUG: ${{ steps.balena-lib.outputs.device_slug }}
+          VERSION: ${{ steps.balena-lib.outputs.os_version }}
+          SOURCE_DIR: ${{ runner.temp }}/deploy
+        run: |
+          if [ -n "$($S3_CMD ls ${S3_URL}/${SLUG}/${VERSION}/)" ] && [ -z "$($S3_CMD ls ${S3_URL}/${SLUG}/${VERSION}/IGNORE)" ]; then
+            echo "::warning::Deployment already exists at ${S3_URL}/${VERSION}"
+            exit 0
+          fi
+
+          echo "${VERSION}" > "${SOURCE_DIR}/${SLUG}/latest"
+          touch "${SOURCE_DIR}/${SLUG}/${VERSION}/IGNORE"
+
+          $S3_CMD del -rf "${S3_URL}/${SLUG}/${VERSION}"
+          $S3_CMD put "${SOURCE_DIR}/${SLUG}/${VERSION}/IGNORE" "${S3_URL}/${SLUG}/${VERSION}/"
+          $S3_CMD ${S3_SYNC_OPTS} dsync "${SOURCE_DIR}/${SLUG}/${VERSION}/" "${S3_URL}/${SLUG}/${VERSION}/"
+          $S3_CMD put "${SOURCE_DIR}/${SLUG}/latest" "${S3_URL}/${SLUG}/" --API-ACL=public-read -f
+          $S3_CMD del "${S3_URL}/${SLUG}/${VERSION}/IGNORE"
+
+      # TODO: consider using ready made s3 actions 
+      # # https://github.com/jakejarvis/s3-sync-action
+      # - uses: jakejarvis/s3-sync-action@v0.5.1
+      #   if: inputs.deploy-s3 == true && steps.balena-lib.outputs.deploy_artifact != 'docker-image'
+      #   with:
+      #     args: --follow-symlinks --sse=AES256 --acl=${{ steps.s3-acl-private.outputs.string || steps.s3-acl-public.outputs.string }}
+      #   env:
+      #     AWS_REGION: ${{ vars.AWS_REGION || vars.S3_REGION }}
+      #     AWS_S3_BUCKET: ${{ vars.AWS_S3_BUCKET || vars.S3_BUCKET }}
+      #     AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID || secrets.S3_SECRET_KEY}}
+      #     AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY || secrets.S3_ACCESS_KEY }}
+      #     SOURCE_DIR: ${{ runner.temp }}/deploy/image
+      #     DEST_DIR: ${{ steps.s3-images-dir.outputs.string || steps.s3-esr-images-dir.outputs.string }}
+
+      # # https://github.com/keithweaver/aws-s3-github-action
+      # - uses: keithweaver/aws-s3-github-action@v1.0.0
+      #   if: inputs.deploy-s3 == true && steps.balena-lib.outputs.deploy_artifact != 'docker-image'
+      #   with:
+      #     command: put
+      #     source: ${{ runner.temp }}/deploy/image/${{ steps.balena-lib.outputs.device_slug }}/latest
+      #     destination: s3://${{ vars.AWS_S3_BUCKET || vars.S3_BUCKET }}/${{ steps.s3-images-dir.outputs.string || steps.s3-esr-images-dir.outputs.string }}/
+      #     aws_access_key_id: ${{ secrets.AWS_SECRET_ACCESS_KEY || secrets.S3_SECRET_KEY}}
+      #     aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY || secrets.S3_ACCESS_KEY }}
+      #     aws_region: ${{ vars.AWS_REGION || vars.S3_REGION }}
```

Comment:
delete these experimental actions, we are replacing S3 upload with webresources anyway